tests/test_qsimh_device.py

```diff
@@ -0,0 +1,400 @@
+            (qml.RX(1.4, wires=[0]).inv(), [cirq.rx(-1.4)]),
+            (qml.RX(-1.2, wires=[0]).inv(), [cirq.rx(1.2)]),
+            (qml.RX(2, wires=[0]).inv(), [cirq.rx(-2)]),
+            (qml.RY(1.4, wires=[0]), [cirq.ry(1.4)]),
+            (qml.RY(0, wires=[0]), [cirq.ry(0)]),
+            (qml.RY(-1.3, wires=[0]), [cirq.ry(-1.3)]),
+            (qml.RY(1.4, wires=[0]).inv(), [cirq.ry(-1.4)]),
+            (qml.RY(0, wires=[0]).inv(), [cirq.ry(0)]),
+            (qml.RY(-1.3, wires=[0]).inv(), [cirq.ry(+1.3)]),
+            (qml.RZ(1.4, wires=[0]), [cirq.rz(1.4)]),
+            (qml.RZ(-1.1, wires=[0]), [cirq.rz(-1.1)]),
+            (qml.RZ(1, wires=[0]), [cirq.rz(1)]),
+            (qml.RZ(1.4, wires=[0]).inv(), [cirq.rz(-1.4)]),
+            (qml.RZ(-1.1, wires=[0]).inv(), [cirq.rz(1.1)]),
+            (qml.RZ(1, wires=[0]).inv(), [cirq.rz(-1)]),
+            (qml.Rot(1.4, 2.3, -1.2, wires=[0]), [cirq.rz(1.4), cirq.ry(2.3), cirq.rz(-1.2)],),
+            (qml.Rot(1, 2, -1, wires=[0]), [cirq.rz(1), cirq.ry(2), cirq.rz(-1)]),
+            (qml.Rot(-1.1, 0.2, -1, wires=[0]), [cirq.rz(-1.1), cirq.ry(0.2), cirq.rz(-1)],),
+            (
+                qml.Rot(1.4, 2.3, -1.2, wires=[0]).inv(),
+                [cirq.rz(1.2), cirq.ry(-2.3), cirq.rz(-1.4)],
+            ),
+            (qml.Rot(1, 2, -1, wires=[0]).inv(), [cirq.rz(1), cirq.ry(-2), cirq.rz(-1)],),
+            (qml.Rot(-1.1, 0.2, -1, wires=[0]).inv(), [cirq.rz(1), cirq.ry(-0.2), cirq.rz(1.1)],),
+            (
+                qml.QubitUnitary(np.array([[1, 0], [0, 1]]), wires=[0]),
+                [cirq.MatrixGate(np.array([[1, 0], [0, 1]]))],
+            ),
+            (
+                qml.QubitUnitary(np.array([[1, 0], [0, -1]]), wires=[0]),
+                [cirq.MatrixGate(np.array([[1, 0], [0, -1]]))],
+            ),
+            (
+                qml.QubitUnitary(np.array([[-1, 1], [1, 1]]) / math.sqrt(2), wires=[0]),
+                [cirq.MatrixGate(np.array([[-1, 1], [1, 1]]) / math.sqrt(2))],
+            ),
+            (
+                qml.QubitUnitary(np.array([[1, 0], [0, 1]]), wires=[0]).inv(),
+                [cirq.MatrixGate(np.array([[1, 0], [0, 1]])) ** -1],
+            ),
+            (
+                qml.QubitUnitary(np.array([[1, 0], [0, -1]]), wires=[0]).inv(),
+                [cirq.MatrixGate(np.array([[1, 0], [0, -1]])) ** -1],
+            ),
+            (
+                qml.QubitUnitary(np.array([[-1, 1], [1, 1]]) / math.sqrt(2), wires=[0]).inv(),
+                [cirq.MatrixGate(np.array([[-1, 1], [1, 1]]) / math.sqrt(2)) ** -1],
+            ),
+        ],
+    )
+    def test_apply_single_wire(self, qsimh_device_1_wire, gate, expected_cirq_gates):
+        """Tests that apply adds the correct gates to the circuit for single-qubit gates."""
+
+        qsimh_device_1_wire.reset()
+
+        # try:
+        qsimh_device_1_wire.apply([gate])
+        # except:
+            # sdfsd
```

Comment:
```suggestion
        qsimh_device_1_wire.apply([gate])
```

Reply:
Another :facepalm:. I find this to be an easy way to get the tests to stop and enter the terminal for debugging. There's probably a better way, but this generally works quite nicely. :laughing: